types/systemjs/index.d.ts

```diff
@@ -1,388 +1,104 @@
-         */
-        resolveSync(moduleName: string, parentName?: string): string;
-
-        /**
-         * In CommonJS environments, SystemJS will substitute the global require as needed by the module format being
-         * loaded to ensure the correct detection paths in loaded code.
-         * The CommonJS require can be recovered within these modules from System._nodeRequire.
-         */
-        _nodeRequire(dep: string): any;
-
-        /**
-         * Modules list available only with trace=true
-         */
-        loads: PackageList<any>;
-
-        env: string;
-
-        loaderErrorStack: boolean;
-
-        packageConfigPaths: string[];
-
-        /**
-         * Specify a value of true to have SystemJS conform to the AMD-style plugin syntax, e.g. "text!some/file.txt", over the default of "some/file.txt!text".
-         */
-        pluginFirst: boolean;
-
-        version: string;
-
-        /**
-         * Enables the output of warnings to the console, including deprecation messages.
-         */
-        warnings: boolean;
-    }
+// TypeScript Version: 3.5
+
+declare namespace System {
+  export {};
+
+  /**
+   * Loads a javascript module from either a url or bare specifier that is in an import map.
+   * You may optionally provide a parentUrl that will be used for resolving relative urls.
+   */
+  // tslint:disable-next-line no-unnecessary-generics
+  export const import: ImportFn;
+
+  /**
+   * Inserts a new module into the SystemJS module registry. The System.register format is
+   * the underlying implementation that allows for ESM emulation.
+   * See https://github.com/systemjs/systemjs/blob/master/docs/system-register.md for more details.
+   * Register may be called with a name argument if you are using the named-register extra. (See
+   * https://github.com/systemjs/systemjs#extras).
+   */
+  export function register(dependencies: string[], declare: DeclareFn): void;
+  export function register(name: string, dependencies: string[], declare: DeclareFn): void;
+
+  /**
+   * Resolve any moduleId to its full URL. For a moduleId that is in the import map, this will resolve
+   * the full import map URL. For a moduleId that is a relative url, the returned url will be resolved
+   * relative to the parentUrl or the current browser page's base url. For a full url, resolve() is
+   * a no-op.
+   */
+  export function resolve(moduleId: string, parentUrl?: string): string;
+
+  /**
+   * Delete a module from the module registry. Note that the moduleId almost always must be a full url and that
+   * you might need to call System.resolve() to obtain the moduleId for modules in an import map.
+   * The returned function is intended for use after re-importing the module. Calling the function
+   * will re-bind all the exports of the re-imported module to every module that depends on the module.
+   */
+  export function delete(moduleId: string): false | UpdateModuleFn;
```

Comment:
Typescript fails to compile this because it thinks it's invalid to export a thing called delete. Any ideas on how to fix?